resources.tf

```diff
@@ -106,3 +106,111 @@ resource "aws_s3_bucket_policy" "published_bucket_https_only" {
   bucket = aws_s3_bucket.published.id
   policy = data.aws_iam_policy_document.published_bucket_https_only.json
 }
+
+resource "aws_iam_role" "dataset_generator" {
+  name               = "dataset_generator"
+  assume_role_policy = data.aws_iam_policy_document.ec2_assume_role.json
+  tags               = local.tags
+}
+
+resource "aws_iam_instance_profile" "dataset_generator" {
+  name = "dataset_generator"
+  role = aws_iam_role.dataset_generator.id
+}
+
+data "aws_iam_policy_document" "ec2_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+#        Attach AWS policies
+resource "aws_iam_role_policy_attachment" "emr_for_ec2_attachment" {
+  role       = aws_iam_role.dataset_generator.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonElasticMapReduceforEC2Role"
+
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_for_ssm_attachment" {
+  role       = aws_iam_role.dataset_generator.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2RoleforSSM"
+
+}
+
+#        Create and attach custom policy
+data "aws_iam_policy_document" "write_s3_dev" {
```

Comment:
Can we name this `analytical_dataset_write_s3` please

Reply:
Could you also add a TODO pointing to DW-3618 please? That will cover off locking this role down to defined resources.